Jenkinsfile

```diff
@@ -0,0 +1,40 @@
+#!groovy
+
+def nodeLabel = 'docker'
+
+/* Only keep the 10 most recent builds. */
+properties([[$class: 'BuildDiscarderProperty',
+                strategy: [$class: 'LogRotator', numToKeepStr: '10']]])
+
+node(nodeLabel) {
+  stage 'Checkout'
+  checkout scm
+
+  stage 'Build'
+
+  /* Call the maven build. */
+  mvn "clean install -B"
+
+  /* Archive the test results */
+  step([$class: 'JUnitResultArchiver', testResults: '**/target/surefire-reports/TEST-*.xml'])
+
+  /* Archive the build artifacts */
+  step([$class: 'ArtifactArchiver', artifacts: 'target/*.hpi'])
+}
+
+/* Run maven from tool "mvn" */
+void mvn(def args) {
+    /* Get jdk tool. */
+    String jdktool = tool name: "jdk8", type: 'hudson.model.JDK'
```

Comment:
I think that we would want the Jenkinsfile to build with jdk7, rather than jdk8, since the release builds are performed with jdk7.  We might want to perform the build twice, once with jdk7 (as the plugin is released from a build based on JDK7) and once with jdk8 (as a safety check to detect if there are problems with jdk8).